### PR TITLE
v1.13 Backports 2023-04-12

### PR DESCRIPTION
**Pull request by jibi**
* [x] #24449 (@MrFreezeex)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 24449; do contrib/backporting/set-labels.py $pr done 1.13; done
```
or with
```
make add-labels BRANCH=v1.13 ISSUES=24449
```

### CODE REVIEW

**Comment by jibi** [comment]
/test-backport-1.13

File: pkg/egressgateway/manager_privileged_test.go [hunk 1]
```diff
@@ -61,11 +61,13 @@ var (
...
 
-	nodeGroup1Labels = map[string]string{"label1": "1"}
-	nodeGroup2Labels = map[string]string{"label2": "2"}
+	nodeGroupNotFoundLabels = map[string]string{"label1": "notfound"}
+	nodeGroup1Labels        = map[string]string{"label1": "1"}
+	nodeGroup2Labels        = map[string]string{"label2": "2"}
 
-	nodeGroup1Selector = &slimv1.LabelSelector{MatchLabels: nodeGroup1Labels}
-	nodeGroup2Selector = &slimv1.LabelSelector{MatchLabels: nodeGroup2Labels}
+	nodeGroupNotFoundSelector = &slimv1.LabelSelector{MatchLabels: nodeGroupNotFoundLabels}
```
Comment (MrFreezeex) [inline pkg/egressgateway/manager_privileged_test.go:68]: I think you dropped a test here that could have been kept AFAICT (or if too cumbersome maybe those change to the labels/labels selectors should probably be dropped as well).

**Review by MrFreezeex — COMMENTED** [review]
Wow didn't realize that excluded cidr was not 1.13 thanks for doing the backport :sweat_smile: :pray:.

I have a small comment about the test in manager_priviledged_test but apart from that it looks good to me :+1:.

File: pkg/egressgateway/manager_privileged_test.go (hunk 1 above)
Comment (jibi) [reply pkg/egressgateway/manager_privileged_test.go:68]: I dropped those changes as they were related to the excluded CIDRs test cases, I'll drop also the unused variables to make the linter happy :+1:

**Comment by jibi** [comment]
> Wow didn't realize that excluded cidr was not 1.13

yeah I mixed things up, I was sure we backported excluded CIDRs to `cilium/cilium` v1.13 but that's clearly not the case :sweat: 

To avoid dropping some bpf tests I synced them from master (minus the excluded CIDRs bits) in https://github.com/cilium/cilium/pull/24859, I'll rebase this on top of those changes

**Comment by jibi** [comment]
/test-backport-1.13

**Review by MrFreezeex — APPROVED** [review]

**Comment by jibi** [comment]
/test-backport-1.13

**Comment by jibi** [comment]
/test-backport-1.13

**Comment by gandro** [comment]
- 'Cilium-PR-K8s-1.16-kernel-4.9' failure is expected, we don't support Kernel 4.9 on Cilium v1.13 anymore

**Review by julianwiedmann — APPROVED** [review]
:+1: :+1:

**Review by dylandreimerink — APPROVED** [review]

**Pull request merged**